src/coinjoin/context.cpp

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2023 The Dash Core developers
+// Distributed under the MIT/X11 software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <coinjoin/context.h>
+
+#include <net.h>
+#include <policy/fees.h>
+#include <txmempool.h>
+
+#ifdef ENABLE_WALLET
+#include <coinjoin/client.h>
+#endif // ENABLE_WALLET
+#include <coinjoin/server.h>
+
+CJContext::CJContext(CConnman& connman, CTxMemPool& mempool, const CMasternodeSync& mn_sync, bool relay_txes) {
+    server = std::make_shared<CCoinJoinServer>(mempool, connman, mn_sync);
+    ::coinJoinServer = server;
+
+#ifdef ENABLE_WALLET
+    clientman = std::make_shared<CJClientManager>(connman, mempool, mn_sync);
+    ::coinJoinClientManagers = clientman;
+
+    if (relay_txes) {
+        queueman = std::make_shared<CCoinJoinClientQueueManager>(connman, mn_sync);
+        ::coinJoinClientQueueManager = queueman;
+    }
+#endif // ENABLE_WALLET
+}
+
+CJContext::~CJContext() {
+#ifdef ENABLE_WALLET
+    ::coinJoinClientQueueManager.reset();
```

Comment:
Check constructor/destructor of LLMQContext and avoid manual creating all members -> you need to control their desctuctions also + can't specify `const` otherwise.

Reply:
Resolved in latest push